### PR TITLE
Generate CodePen buttons in templates instead of on page load.

### PR DESCRIPTION
**Pull request by nicolasgarnier**
r: @addyosmani

### CODE REVIEW

**Comment by surma** [comment]
Code LGTM. I’ll let @addyosmani the final call on performance. Quick test in IE11 showed that we are still pending 1.7s in appendChild().

**Comment by addyosmani** [comment]
Testing out now!

File: docs/_assets/codepen.js [hunk 1]
```diff
@@ -36,29 +36,12 @@ CodeBlockCodePen.prototype.MDLIBS = [
 CodeBlockCodePen.prototype.init = function() {
   'use strict';
 
-  for (var i = 0, len = this.htmlCodeBlocks.length; i < len; i++) {
-    var pre = this.htmlCodeBlocks[i];
+  for (var i = 0, len = this.codepenButtons.length; i < len; i++) {
```
Comment (samccone) [inline docs/_assets/codepen.js]: eh :)

```js
Array.prototype.slice.call(this.codepenButtons).forEach(function(...) { ... })
```

Comment (samccone) [inline docs/_assets/codepen.js]: @sindresorhus will yell at me about Array prototype :joy_cat: so 

```js
[].slice.call(this.codepenButtons).forEach(function(...) { ... })
```

will also work :+1:

Comment (surma) [inline docs/_assets/codepen.js]: Personally, I’m still torn which one these versions I should use.

Comment (Garbee) [inline docs/_assets/codepen.js]: Why would he yell? I thought `Array.prototype` (from reading online, not testing so don't kill me) caused no needless object to get created where `[]` will create an object that essentially does nothing.  (Ok, *one* needless object not a big deal but still...)

Comment (samccone) [inline docs/_assets/codepen.js]: just say whatever to old browsers and use

for(let node of this.codepenButtons) { ... } and make @sebmck happy

Comment (nicolasgarnier) [inline docs/_assets/codepen.js]: Done :D

**Comment by nicolasgarnier** [comment]
Done PTAL @addyosmani :)

**Comment by addyosmani** [comment]
Seeing improvements over the previous way we were handling this. LGTM.

**Comment by samccone** [comment]
:hand:

**Pull request closed without merging**